Add license data to recipes

ViridianMelody opened the pull request:
Small modification to jinja template to pull license data per model from json.

Since it has about 80 outputs, this recipe will need to be built and uploaded manually after merging.

ViridianMelody committed:
Add license to jinja template

ViridianMelody committed:
Remove _spacy_models_repo

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_build_help.py", line 555, in check_recipe
    o = int(path.split("/")[1])
        ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'host'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_956hfa10vj/clone` failed. (See above for error)
</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_build_help.py", line 555, in check_recipe
    o = int(path.split("/")[1])
        ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'host'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_686i8zplt6/clone` failed. (See above for error)
</details>

ViridianMelody committed:
Bump build number

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_build_help.py", line 555, in check_recipe
    o = int(path.split("/")[1])
        ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'host'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_98vwdgdhm6/clone` failed. (See above for error)
</details>

ViridianMelody committed:
Rerun _update_spacy_recipe.py w/ new build number

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_build_help.py", line 555, in check_recipe
    o = int(path.split("/")[1])
        ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'host'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_7fkh5cl1ao/clone` failed. (See above for error)
</details>

skupr-anaconda reviewed (commented):
Overall looks good but could you try to replace the license names as SPDX (https://spdx.org/licenses/):
`GNU GPL 3.0` -> `GPL-3.0-or-later`
`CC BY-SA 3.0` -> `CC-BY-SA-3.0`
`CC BY-SA 4.0` -> `CC-BY-SA-4.0`
`LGPL-LR` -> `LGPLLR`

ViridianMelody committed:
Translate license names to be SPDX compliant

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_build_help.py", line 555, in check_recipe
    o = int(path.split("/")[1])
        ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'host'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_0frzfz01xr/clone` failed. (See above for error)
</details>

ViridianMelody commented:
> Overall looks good but could you try to replace the license names as SPDX (https://spdx.org/licenses/): `GNU GPL 3.0` -> `GPL-3.0-or-later` `CC BY-SA 3.0` -> `CC-BY-SA-3.0` `CC BY-SA 4.0` -> `CC-BY-SA-4.0` `LGPL-LR` -> `LGPLLR`

@skupr-anaconda I've updated the script accordingly, and added `CC BY-NC-SA 3.0` -> `CC-BY-NC-SA-3.0` to your list. Since they're not following SPDX standards, we'll need to add any future licenses to the `LICENSES` dict in `_update_spacy_recipe.py`.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_build_help.py", line 555, in check_recipe
    o = int(path.split("/")[1])
        ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'host'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_42w5r_ng4o/clone` failed. (See above for error)
</details>
